pages/common/aws-s3api.md

```diff
@@ -0,0 +1,36 @@
+# aws s3api
+
+> Create and delete Amazon S3 buckets, and edit bucket properties.
+> More information: <https://awscli.amazonaws.com/v2/documentation/api/latest/reference/s3api/index.html>.
+
+- Create a bucket:
+
+`aws s3api create-bucket --bucket my-bucket`
+
+- Delete a bucket:
+
+`aws s3api delete-bucket --bucket {{bucket_name}}`
+
+- List buckets:
+
+`aws s3api list-buckets`
+
+- List the objects inside of a bucket, and only show each object's key and size:
+
+`aws s3api list-objects --bucket {{bucket_name}} --query '{{Contents[].{Key: Key, Size: Size}}}'`
+
+- Add an object to a bucket:
+
+`aws s3api put-object --bucket {{bucket_name}} --key {{object_key}} --body {{path/to/file}}`
+
+- Download object from a bucket (The output file is always the last argument):
+
+`aws s3api get-object --bucket {{bucket_name}} --key {{object_key}} {{path/to/output_file}}`
+
+- Apply an Amazon S3 bucket policy to a specified bucket:
+
+`aws s3api put-bucket-policy --bucket {{bucket_name}} --policy file://{{path/to/bucket_policy.json}}`
```

Comment:
is `--policy {{smb://path/to/...}}` an alternative?

Reply:
`smb` and `file` are fundamentally similar, and IMO there's not much need to specify `smb:`.

Reply:
@bl-ue just out of curiousity, have you managed to get this command to work with smb? Whenever I do it, aws gets confused.

Reply:
Hmm no, I can't get it to work either.
```
An error occurred (MalformedPolicy) when calling the PutBucketPolicy operation: Policies must be valid JSON and the first byte must be '{'
```

Reply:
In that case we must assume it is not compatible with aws s3api.

Reply:
I agree — there are likely a very few users of `s3api` + `smb:` + `tldr` 😆

Reply:
If I'm not mistaken, for security reasons, smb should generally be avoided anyways.